packages/tools/kda-cli/src/pact/l2.ts

```diff
@@ -0,0 +1,88 @@
+import * as coin from './coin.js';
+import { Reducer } from './pact.js';
+type Decimal = `${number}.${number}`;
+interface IGasCapability {
+  name: 'n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.gas-station.GAS_PAYER';
+  args: [string, { int: number }, number];
+  signer: string;
+}
+
+interface ITransferCapability {
+  name: 'n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.l2.TRANSFER';
+  args: [string, string, number];
+  signer: string;
+}
+
+interface IWithdrawCapability {
+  name: 'n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.l2.WITHDRAW';
+  args: [string, string, number];
+  signer: string;
+}
+
+export interface ICapabilities {
+  Gas: IGasCapability;
+  Transfer: ITransferCapability;
+  Withdraw: IWithdrawCapability;
+}
+
+export interface ITransactionPayload {
+  command: `(n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.l2.transfer "${string}" "${string}" ${Decimal})`;
+  caps: [ITransferCapability, coin.ICapabilities['Gas'], IGasCapability];
+  data: unknown;
+}
+
+export interface IDepositPayload {
+  command: `(n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.l2.deposit "${string}" "${string}" ${Decimal})`;
+  caps: [
+    coin.ICapabilities['Transfer'],
+    coin.ICapabilities['Gas'],
+    IGasCapability,
+  ];
+  data: unknown;
```

Comment:
You could make this a generic. Not sure if necessary

(same for other occurrences of `data: unknown`)

```suggestion
export interface IDepositPayload<T> {
  command: `(n_560eefcee4a090a24f12d7cf68cd48f11d8d2bd9.l2.deposit "${string}" "${string}" ${Decimal})`;
  caps: [
    coin.ICapabilities['Transfer'],
    coin.ICapabilities['Gas'],
    IGasCapability,
  ];
  data: T;
```

Reply:
Not for now, when this function gets added to the client, it should be though.